lib/src/merge.rs

```diff
@@ -68,6 +68,14 @@ impl<T> Diff<T> {
         }
     }
 
+    /// Combine a `Diff<T>` and a `Diff<U>` into a `Diff<(T, U)>`.
+    pub fn zip<U>(self, other: Diff<U>) -> Diff<(T, U)> {
+        Diff {
+            before: (self.before, other.before),
+            after: (self.after, other.after),
+        }
+    }
```

Comment:
nit: Add a test case. I know it's trivially correct, and it's tested indirectly by other tests, but it's also trivial to add a test case.